cmd/tfchainc/erc20cmd.go

```diff
@@ -0,0 +1,100 @@
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"os"
+	"reflect"
+
+	"github.com/spf13/cobra"
+	"github.com/threefoldfoundation/tfchain/cmd/tfchainc/internal"
+	"github.com/threefoldtech/rivine/pkg/cli"
+)
+
+// createERC20Cmd creates rootcommand for ERC20 and adds a subcommand
+// if rootcommand executed the user will also see the output of the syncing status of ethereum
+func createERC20Cmd(client *internal.CommandLineClient) *cobra.Command {
+	erc20SubCmds := &erc20SubCmds{cli: client}
+
+	// define Rootcommand
+	var (
+		rootCmd = &cobra.Command{
+			Use:   "erc20",
+			Short: "Perform erc20 actions",
+			Long:  "Perform erc20 actions",
+			Run:   erc20SubCmds.getSyncingStatus,
+		}
+		getSyncingStatusCmd = &cobra.Command{
+			Use:   "syncstatus",
+			Short: "Get the ethereum sync status",
+			Long:  `Get the ethereum chain sync status.`,
+			Run:   erc20SubCmds.getSyncingStatus,
+		}
+	)
+
+	rootCmd.AddCommand(getSyncingStatusCmd)
+
+	// register flags
+	getSyncingStatusCmd.Flags().Var(
+		cli.NewEncodingTypeFlag(0, &erc20SubCmds.getSyncingStatusCfg.EncodingType, 0), "encoding",
+		cli.EncodingTypeFlagDescription(0))
+
+	return rootCmd
+}
+
+type erc20SubCmds struct {
+	cli                 *internal.CommandLineClient
+	getSyncingStatusCfg struct {
+		EncodingType cli.EncodingType
+	}
+}
+
+func (erc20SubCmds *erc20SubCmds) getSyncingStatus(cmd *cobra.Command, args []string) {
+	erc20cmds := internal.NewERC20Client(erc20SubCmds.cli)
```

Comment:
There is honestly no need to make an entire client here. I would simply call the endpoint directly here. You're adding an abstraction layer here without any benefit.

Reply:
Changed in: https://github.com/threefoldfoundation/tfchain/pull/307/commits/bdce1a1b3d82ccecd576e9aa8f99c92d3f270699